src/mokernels/moinput.jl

```diff
@@ -1,3 +1,5 @@
+abstract type AbstractMOInput{T} <: AbstractVector{T} end
```

Comment:
What are the pros/cons of including the tuple construct in the abstract type parameterisation? Alternatively we could also have
```suggestion
abstract type AbstractMOInput{T} <: AbstractVector{Tuple{T,Int}} end
```

Reply:
Sure, I am fine with that. I started with 
```julia 
abstract type AbstractMOInputabstract{Tuple{T,Int}} <: AbstractVector{Tuple{T,Int}} end
```
which does not work, so I went very simple to get it to work.

Reply:
Tried the suggestion, it does precompile, but then it fails the tests, as `collect` no longer knows what to do.